resources/testharness.js

```diff
@@ -1771,20 +1773,25 @@
     /**
      * Assert that ``actual`` is a number less than ``expected``.
      *
-     * @param {number} actual - Test value.
-     * @param {number} expected - Number that ``actual`` must be less than.
+     * @param {number|bigint} actual - Test value.
+     * @param {number|bigint} expected - Value that ``actual`` must be less than.
      * @param {string} [description] - Description of the condition being tested.
      */
     function assert_less_than(actual, expected, description)
     {
         /*
-         * Test if a primitive number is less than another
+         * Test if a primitive number (or bigint) is less than another
          */
-        assert(typeof actual === "number",
+        assert(typeof actual === "number" || typeof actual === "bigint",
                "assert_less_than", description,
                "expected a number but got a ${type_actual}",
                {type_actual:typeof actual});
 
+        assert(typeof actual === typeof expected,
```

Comment:
Doesn't comparison between numbers and bigints work, or why must the types match?

Reply:
If I change Gecko to make some API return bigints instead of numbers, I expect tests to fail. testharness.js is usually strict on details like this.